File: src/main/java/com/indiepost/repository/jpa/SettingRepositoryJpa.java
package com.indiepost.repository.jpa;

import com.indiepost.model.Setting;
import com.indiepost.repository.SettingRepository;
import org.springframework.stereotype.Repository;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;

/**
 * Created by jake on 8/31/17.
 */
@Repository
public class SettingRepositoryJpa implements SettingRepository {

    @PersistenceContext
    private EntityManager entityManager;

    @Override
    public Setting get() {
        return entityManager.find(Setting.class, 1L);
    }

    @Override
    public void update(Setting setting) {
        entityManager.merge(setting);
    }
}

File: src/main/kotlin/com/indiepost/repository/jpa/SettingRepositoryJpa.kt
package com.indiepost.repository.jpa

import com.indiepost.model.Setting
import com.indiepost.repository.SettingRepository
import org.springframework.stereotype.Repository

import javax.persistence.EntityManager
import javax.persistence.PersistenceContext

/**
 * Created by jake on 8/31/17.
 */
@Repository
class SettingRepositoryJpa : SettingRepository {

    @PersistenceContext
    private lateinit var entityManager: EntityManager

    override fun get(): Setting {
        return entityManager.find(Setting::class.java, 1L)
    }

    override fun update(setting: Setting) {
        entityManager.merge(setting)
    }
}
